dev-client/src/screens/ProjectTeamScreen/components/UserList.tsx

```diff
@@ -15,165 +15,47 @@
-  Pick<ItemProps, 'memberAction'> & {
-    isCurrentUser: boolean;
-    currentUserRole: ProjectRole;
-  }) =>
-  !isCurrentUser && currentUserRole === 'MANAGER' ? (
-    <Pressable onPress={memberAction}>{children}</Pressable>
-  ) : (
-    <>{children}</>
-  );
-
-function UserItem({
-  membership,
-  user,
-  currentUserId,
-  removeUser,
-  memberAction,
-  currentUserRole,
-}: ItemProps) {
-  const {t} = useTranslation();
-  const isCurrentUser = useMemo(() => {
-    return user.id === currentUserId;
-  }, [user, currentUserId]);
-
-  const userName = useMemo(() => {
-    let name = formatName(user.firstName, user.lastName);
-
-    if (isCurrentUser) {
-      name += ` (${t('general.you')})`;
-    }
-    return name;
-  }, [user, isCurrentUser, t]);
-
-  return (
-    <Box borderBottomWidth="1" width={275} py={2}>
-      <VStack>
-        <UserWrapper
-          currentUserRole={currentUserRole}
-          isCurrentUser={isCurrentUser}
-          memberAction={memberAction}>
-          <HStack space={3} justifyContent="space-between" alignItems="center">
-            <Box>
-              <Image
-                variant="profilePic"
-                source={{uri: user.profileImage}}
-                alt="profile pic"
-              />
-            </Box>
-            <Text flex={3}>{userName}</Text>
-            <Box>
-              <Badge
-                variant="chip"
-                bg="primary.lighter"
-                py="5px"
-                px="10px"
-                _text={{color: 'text.primary'}}>
-                {t('general.role.' + membership.userRole)}
-              </Badge>
-            </Box>
-          </HStack>
-        </UserWrapper>
-        {isCurrentUser && (
-          <ConfirmModal
-            trigger={onOpen => (
-              <LeaveProjectTrigger
-                onOpen={onOpen}
-                message={t('projects.team.leave_project_modal.trigger')}
-              />
-            )}
-            title={t('projects.team.leave_project_modal.title')}
-            body={t('projects.team.leave_project_modal.body')}
-            actionName={t('projects.team.leave_project_modal.action_name')}
-            handleConfirm={removeUser}
-          />
-        )}
-      </VStack>
-    </Box>
-  );
-}
-
 export const UserList = ({
   memberships,
-  userAction,
+  removeUser,
   currentUserId,
   memberAction,
   currentUserRole,
 }: ListProps) => {
+  const hasSingleManager = useMemo(
+    () => memberships.filter(m => m[0].userRole === 'MANAGER').length === 1,
+    [memberships],
+  );
+
   return (
     <FlatList
       data={memberships}
       renderItem={({item: [membership, user]}) => (
         <UserItem
           membership={membership}
           user={user}
-          currentUserId={currentUserId}
-          removeUser={userAction(membership)}
+          isCurrentUser={user.id === currentUserId}
+          isManager={currentUserRole === 'MANAGER'}
+          hasSingleManager={hasSingleManager}
```

Comment:
Suggestion (naming): Also this is about if the project has a single manager, not about this User in particular? If so, maybe name something like "projectHasSingleManager".

I guess with the name UserItem, I think of some of the other props as stuff specific to the "UserItem", but this and the previous prop as more like "context"-type stuff

Reply:
This makes sense to me. I'll see if I can address your and Paul's comments with a more easy-to-understand model here.